### PR TITLE
bpo-39723: make io.open_code accept PathLike objects

### PR DESCRIPTION
**Pull request by hauntsaninja**
https://bugs.python.org/issue39723

### CODE REVIEW

File: Modules/_io/_iomodule.c [hunk 1]
```diff
@@ -499,9 +499,9 @@ with calling open(path, 'rb').
 
 static PyObject *
 _io_open_code_impl(PyObject *module, PyObject *path)
-/*[clinic end generated code: output=2fe4ecbd6f3d6844 input=f5c18e23f4b2ed9f]*/
+/*[clinic end generated code: output=2fe4ecbd6f3d6844 input=9121d12e66f85b25]*/
 {
-    return PyFile_OpenCodeObject(path);
+    return PyFile_OpenCodeObject(PyOS_FSPath(path));
```
Comment (lazka) [inline Modules/_io/_iomodule.c]: Where does the new reference returned by PyOS_FSPath() get freed?

Comment (hauntsaninja) [reply Modules/_io/_iomodule.c]: Thanks, hopefully in this next commit :-)

**Comment by tirkarthi** [comment]
`io.open_code` shouldn't be accepting path-like objects as noted in https://bugs.python.org/msg362522 . I would propose closing this PR and opening another one for documentation update as @zooba suggested.

**Comment by hauntsaninja** [comment]
Sounds good, opened #19824 for the documentation update.

**Pull request closed without merging**